lib/diagnostic.rb

```diff
@@ -46,6 +46,13 @@ def transform(words)
   # replacing the trailing newline character with a
   # space and return the result.
   def read_file(filename)
-    # Your code goes here
+    result = ''
+    File.open(filename) do |file|
+      file.each do |line|
+        result += line.strip + ' '
+      end
+    end
+      puts result
```

Comment:
This didn't pass the tests but it looks like you were on the right track. Check out the solution code if you need a nudge in the right direction.